sdk/azidentity/managed_identity_credential.go

```diff
@@ -54,10 +68,10 @@ func NewManagedIdentityCredential(clientID string, options *ManagedIdentityCrede
 	// Assign the msiType discovered onto the client
 	client.msiType = msiType
 	// check if no clientID is specified then check if it exists in an environment variable
-	if len(clientID) == 0 {
-		clientID = os.Getenv("AZURE_CLIENT_ID")
+	if len(id) == 0 {
+		id = os.Getenv("AZURE_CLIENT_ID")
```

Comment:
Do we also need to support a resource ID env var?

Reply:
I do think we need to, but I wanted to wait to talk to someone like Jon Gallant who I think dealt with all of the env vars that we allow and naming before adding it. I opened an issue to track that: https://github.com/Azure/azure-sdk-for-go/issues/14742